[1.21] internal exception kicks player from server due to message codec failure

https://github.com/FallingColors/HexMod/blob/6e14fae9510895a71b93d8970e91402700a5218a/Common/src/main/java/at/petrak/hexcasting/api/casting/mishaps/MishapInternalException.kt#L19

when causing an internal exception, since the raw exception is passed to Component.translatable, server got something like
```log
[018��2026 13:20:38.836] [Netty Server IO #6/ERROR] [net.minecraft.network.Connection/]: Exception caught in connection
io.netty.handler.codec.EncoderException: Failed to encode packet 'clientbound/minecraft:system_chat'
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.codec.IdDispatchCodec.encode(IdDispatchCodec.java:55) ~[client-1.21.1-20240808.144430-srg.jar%23234!/:?]
[......STACKTRACE OMITTED......]
Caused by: io.netty.handler.codec.EncoderException: Failed to encode: This value needs to be parsed as component translation{key='hexcasting.mishap.unknown', args=[dev.latvian.mods.rhino.EvaluatorException: Can't find method at.petrak.hexcasting.api.casting.eval.vm.CastingImage.copy(at.petrak.hexcasting.api.utils.TreeList$TreeList1,number,at.petrak.hexcasting.api.utils.TreeList$TreeList0,boolean,number,net.minecraft.nbt.CompoundTag). (startup_scripts:HEX/patterns/RegistryBase.js#75)]}
[......STACKTRACE OMITTED......]
```
, which results in an instant kick

--- a/Common/src/main/java/at/petrak/hexcasting/api/casting/mishaps/MishapInternalException.kt
+++ b/Common/src/main/java/at/petrak/hexcasting/api/casting/mishaps/MishapInternalException.kt
@@ -3,7 +3,14 @@ package at.petrak.hexcasting.api.casting.mishaps
 import at.petrak.hexcasting.api.casting.eval.CastingEnvironment
 import at.petrak.hexcasting.api.casting.iota.Iota
 import at.petrak.hexcasting.api.pigment.FrozenPigment
+import at.petrak.hexcasting.api.utils.TreeList
+import net.minecraft.network.chat.ClickEvent
+import net.minecraft.network.chat.Component
+import net.minecraft.network.chat.HoverEvent
+import net.minecraft.network.chat.Style
 import net.minecraft.world.item.DyeColor
+import java.io.PrintWriter
+import java.io.StringWriter
 
 class MishapInternalException(val exception: Exception) : Mishap() {
     override fun accentColor(env: CastingEnvironment, errorCtx: Context): FrozenPigment =
@@ -13,6 +20,21 @@ class MishapInternalException(val exception: Exception) : Mishap() {
         // NO-OP
     }
 
-    override fun errorMessage(env: CastingEnvironment, errorCtx: Context) =
-        error("unknown", exception)
+    override fun errorMessage(ctx: CastingEnvironment, errorCtx: Context): Component {
+        var message = Component.literal("$exception")
+
+        // dump stack trace
+        val sw = StringWriter()
+        val pw = PrintWriter(sw)
+        exception.printStackTrace(pw)
+        val trace = sw.toString()
+
+        message = message.withStyle(
+            Style.EMPTY
+                .withClickEvent(ClickEvent(ClickEvent.Action.COPY_TO_CLIPBOARD, trace))
+                .withHoverEvent(HoverEvent(HoverEvent.Action.SHOW_TEXT, Component.literal(trace)))
+        )
+
+        return error("unknown", message)
+    }
 }

--- a/Common/src/main/java/at/petrak/hexcasting/api/casting/mishaps/MishapInternalException.kt
+++ b/Common/src/main/java/at/petrak/hexcasting/api/casting/mishaps/MishapInternalException.kt
@@ -3,7 +3,6 @@ package at.petrak.hexcasting.api.casting.mishaps
 import at.petrak.hexcasting.api.casting.eval.CastingEnvironment
 import at.petrak.hexcasting.api.casting.iota.Iota
 import at.petrak.hexcasting.api.pigment.FrozenPigment
-import at.petrak.hexcasting.api.utils.TreeList
 import net.minecraft.network.chat.ClickEvent
 import net.minecraft.network.chat.Component
 import net.minecraft.network.chat.HoverEvent

